oncotator/index/TabixIndexer.py

```diff
@@ -123,17 +124,41 @@ def index(destDir, inputFilename, fileColumnNumList=None, preset=None):
             return outputFilename
 
         outputFilename = os.path.join(destDir, string.join([fileName, ".tabix_indexed", fileExtension], ""))
-        # Copy the input file to output file.
-        shutil.copyfile(inputFilename, outputFilename)
 
         # Load the file into a tsvReader.
         if preset in ("gff", "bed", "sam", "vcf", "psltbl", "pileup"):
+            # Copy the input file to output file.
+            shutil.copyfile(inputFilename, outputFilename)
             tabix_index = pysam.tabix_index(filename=outputFilename, force=True, preset=preset)
         else:
-            # Have to specify min_size=0 in pysam 0.8.1 to get pysam to correctly output a .tbi file 
+
+            # Need to comment out the header line with a "#", so we cannot simply copy the file.
+            input_reader = GenericTsvReader(inputFilename)
+
+            output_writer = file(outputFilename, 'w')
+            output_writer.writelines(input_reader.getCommentsAsList())
+
+            # Add "#" for the header line.
+            output_writer.write("#")
+            field_names = input_reader.getFieldNames()
+            output_writer.write("\t".join(field_names))
+            output_writer.write("\n")
+            output_writer.flush()
+
+            # Write the rest of the file
+            # This might be too slow, since a raw reader would be pretty fast.
+            for line_dict in input_reader:
+                line_list = [line_dict[k] for k in field_names]
+                line_rendered = "\t".join(line_list) + "\n"
+                output_writer.write(line_rendered)
+
+            output_writer.close()
```

Comment:
Using ``with`` block for writer
``GenericTsvReader`` does not support being in a ``with`` block (for now)
Added a close method to GenericTsvReader and calling it.